server/etcdserver/api/v3alarm/alarms.go

```diff
@@ -123,15 +111,14 @@ func (a *AlarmStore) restore() error {
 	tx := b.BatchTx()
 
 	tx.Lock()
-	tx.UnsafeCreateBucket(buckets.Alarm)
-	err := tx.UnsafeForEach(buckets.Alarm, func(k, v []byte) error {
-		var m pb.AlarmMember
-		if err := m.Unmarshal(k); err != nil {
-			return err
-		}
-		a.addToMap(&m)
-		return nil
-	})
+	buckets.UnsafeCreateAlarmBucket(tx)
+	ms, err := buckets.UnsafeGetAllAlarms(tx)
+	if err != nil {
+		return err
```

Comment:
Are we missing here tx.Unlock() ? 
probably good reason to move it to a separate method with a 'defer' block

Reply:
Done.